crmsh/sbd.py

```diff
@@ -3,14 +3,265 @@
+        else:
+            adjust_value = value_from_calculation
+        return adjust_value
+
+    @staticmethod
+    def get_suitable_sbd_systemd_timeout():
+        """
+        Get suitable systemd start timeout for sbd.service
+        """
+        return int(1.2 * int(SBDManager.get_sbd_value_from_config("SBD_DELAY_START")))
+
+    @staticmethod
+    def is_delay_start():
+        """
+        Check if SBD_DELAY_START is not no or not set
+        """
+        res = SBDManager.get_sbd_value_from_config("SBD_DELAY_START")
+        return res and res != "no"
+
+    def adjust_systemd_start_timeout(self):
+        """
+        Adjust start timeout for sbd when set SBD_DELAY_START
+        """
+        if not self.two_node_without_qdevice and os.path.isdir(SBD_SYSTEMD_DELAY_START_DIR):
+            shutil.rmtree(SBD_SYSTEMD_DELAY_START_DIR)
+            bootstrap.csync2_update(SBD_SYSTEMD_DELAY_START_DIR)
+            utils.cluster_run_cmd("systemctl daemon-reload")
+            return
+
+        # TimeoutStartUSec default is 1min 30s, need to parse as seconds
+        cmd = "systemctl show -p TimeoutStartUSec sbd --value"
+        out = utils.get_stdout_or_raise_error(cmd)
+        res_seconds = re.search("(\d+)s", out)
+        default_start_timeout = int(res_seconds.group(1)) if res_seconds else 0
+        res_min = re.search("(\d+)min", out)
+        default_start_timeout += 60 * int(res_min.group(1)) if res_min else 0
+        if default_start_timeout >= self.sbd_delay_start_value:
+            return
+
+        utils.mkdirp(SBD_SYSTEMD_DELAY_START_DIR)
+        sbd_delay_start_file = "{}/sbd_delay_start.conf".format(SBD_SYSTEMD_DELAY_START_DIR)
+        utils.str2file("[Service]\nTimeoutSec={}".format(int(1.2*self.sbd_delay_start_value)), sbd_delay_start_file)
+        bootstrap.csync2_update(SBD_SYSTEMD_DELAY_START_DIR)
+        utils.cluster_run_cmd("systemctl daemon-reload")
+
+    def adjust_stonith_timeout(self):
+        """
+        Adjust stonith-timeout property
+        """
+        utils.set_property(stonith_timeout=self._get_stonith_timeout())
+
+    def adjust_pcmk_delay_max(self):
+        """
+        Adjust pcmk_delay_max parameter for sbd ra
+        """
+        # TODO this function should be outside of sbd.py, to adjust any fence device
+
+        if not utils.has_resource_configured(SBDManager.SBD_RA):
+            return
+
+        if self.two_node_without_qdevice:
+            cmd = "crm resource param {} set pcmk_delay_max {}s".format(SBDManager.SBD_RA_ID, self.pcmk_delay_max)
+        else:
+            cmd = "crm resource param {} delete pcmk_delay_max".format(SBDManager.SBD_RA_ID)
+        utils.get_stdout_or_raise_error(cmd)
+
+    def adjust_delay_start(self):
+        """
+        Adjust SBD_DELAY_START in /etc/sysconfig/sbd
+        """
+        value = str(self.sbd_delay_start_value) if self.two_node_without_qdevice else "no"
```

Comment:
Sorry, SBD_DELAY_START is only matters to virtualization environment no matter diskless or disk-based, no matter the cluster size or qdevice.

The better name is adjust_sbd_delay_start() to make it consistent to get_sbd_delay_start()

Reply:
Changed as:
```
def adjust_sbd_delay_start(self):
        """
        Adjust SBD_DELAY_START in /etc/sysconfig/sbd
        """
        value_from_configuration = SBDManager.get_sbd_value_from_config("SBD_DELAY_START")
        value_from_calculation = str(self.sbd_delay_start_value) if utils.detect_virt() else "no"
        if value_from_configuration != value_from_calculation:
            SBDManager.update_configuration({"SBD_DELAY_START": value_from_calculation})
```